### PR TITLE
i18-rails-and-blacklight-upgrade

### PR DESCRIPTION
**Pull request by aprilrieger**
upgrade Rails to 6.1
upgrade Blacklight gem to 7.33.1

branch: i18-rails-upgrade-6point1
get `.env.development` here: https://start.1password.com/open/i?a=LTLZ652TT5H5FHMYMASSH7PIXM&v=huuakin4bu4xanlhktv42qheam&i=ok6rwracafccnhrhur7usyfwoi&h=scientist.1password.com

```yaml
gem install stack_car -v0.16.0
gem install dory

dory up
sc build
sc up
```
navigate to `http://oralhistory.test/users/sign_in`
login credentials are in the .env.development <---- stack_car will make the values in .env.development override what is in the .env file. The .env.development file should be greyed out in VS Code, listed in the .gitignore, and should never be committed, shhh they are secrets!

navigate to `http://oralhistory.test/admin` &&

run importer // OR // run the importer in a running web container `rake db:migrate import[10]` [#] <— replace "#" with amount of works you want to import.

Need troubleshooting tips?
If you do a `dc down -v`, odds are you will want to do a `rm -rf tmp/*`

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -22,7 +22,9 @@ services:
   web:
     build:
       context: .
-      dockerfile: Dockerfile
+      target: web
+      cache_from:
+        - "${REGISTRY_HOST}${REGISTRY_URI}/oral-history:${TAG}"
```
Comment (aprilrieger) [inline docker-compose.yml:27]: Adding to help speed up the build process. Not sure how to tell if it has or not?

File: solr/conf/schema.xml [hunk 2]
```diff
@@ -315,7 +314,6 @@
     <fieldType class="solr.TextField" name="textSuggest" positionIncrementGap="100">
       <analyzer>
         <tokenizer class="solr.KeywordTokenizerFactory"/>
-        <filter class="solr.StandardFilterFactory"/>
```
Comment (aprilrieger) [inline solr/conf/schema.xml:318]: This is getting taken out in Solr 8. I was trying my hand at getting that across the line last night/this morning but it wasn't working so I cut my losses. I did however forget to add this back in and can if you want.

File: config/application.rb [hunk 3]
```diff
@@ -19,11 +22,18 @@
 module OralHistory
   class Application < Rails::Application
     # Initialize configuration defaults for originally generated Rails version.
-    config.load_defaults 5.1
+    config.load_defaults 6.1
+    config.autoloader = :classic
```
Comment (orangewolf) [inline config/application.rb:26]: does classic mean spring here? I think that's fine, but we might sneak in zeitwork at some point

File: docker-compose.yml [hunk 4]
```diff
@@ -48,8 +50,7 @@ services:
   worker:
     build:
       context: .
-      dockerfile: Dockerfile
-    image: "${REGISTRY_HOST}${REGISTRY_URI}/oral-history:${TAG}"
+      target: worker
```
Comment (orangewolf) [inline docker-compose.yml:53]: no image here? its a different image now, so I think you need one

File: solr/conf/schema.xml (hunk 2 above)
Comment (orangewolf) [reply solr/conf/schema.xml:318]: I think you have to if you want search to work. it won't tokenize properly in 7 otherwise

**Review by orangewolf — CHANGES_REQUESTED** [review]
only think I think we really need are the docker-compose worker image and the solr schema.xml fix (yours might work right now because you haven't down -v and up'd)

**Review by orangewolf — APPROVED** [review]

**Pull request merged**